### ISSUE TITLE
Grammar enhancement: parsing var indexing

### ISSUE THREAD

**Issue by kamadorueda** [amplify-education/python-hcl2#42]
While parsing

```tf
resource "aws_route53_record" "web-ephemeral-certificate-validation" {
  name    = aws_acm_certificate.web-ephemeral-certificate.domain_validation_options.0.resource_record_name
  type    = aws_acm_certificate.web-ephemeral-certificate.domain_validation_options.0.resource_record_type
  zone_id = data.aws_route53_zone.fluidattacks.id
  records = [aws_acm_certificate.web-ephemeral-certificate.domain_validation_options.0.resource_record_value]
  ttl     = 60
}
```

It fails at `xx.0.xx` which is valid HCL2 syntax (and real terraform production code):

```
lark.exceptions.UnexpectedToken: Unexpected token Token(DECIMAL, '0') at line 2, column 85.
```

**Comment by holgiosalos** [comment]
Hi, this PR https://github.com/amplify-education/python-hcl2/pull/36 fixes that problem, but it is pending from approval since July.

**Comment by aoskotsky-amplify** [comment]
Should be fixed now with #44. Let me know if there are any issues
